### ISSUE TITLE
(kubernetes) add properties for affinity specs for pod templates

### ISSUE THREAD

**Issue by Barbiero** [quarkusio/quarkus#13596]
**Description**
More info: [Affinity and anti-affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity)

Sometimes it's useful to set certain pods to have affinity or anti-affinity for the nodes in your cluster. It would be nice if we could use properties like `quarkus.kubernetes.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[]` and similar ones

Currently, I can use affinities by creating a `src/main/kubernetes/kubernetes.yaml` file with the affinity defined in it, but it's a bit troublesome because I'd like to define different affinities according to the `quarkus.profile` used at build time.

**Comment by geoand** [comment]
@iocanel Does dekorate support these?

**Comment by iocanel** [comment]
I don't think that it does. Let's create a dekorator for that on Quarkus and we can remove it, when dekorate provides one

**Comment by geoand** [comment]
@iocanel do you have the cycles to do this? I know I don't :)
If you don't as well, let's find someone else to work on this :)

**Comment by iocanel** [comment]
I'll do it!

**Comment by geoand** [comment]
Thanks!
